core/templates/dev/head/pages/exploration_editor/ExplorationDataService.js

```diff
@@ -103,31 +105,37 @@ oppia.factory('ExplorationDataService', [
           // changes applied. This makes a force-refresh necessary when changes
           // are discarded, otherwise the exploration-with-draft-changes
           // (which is cached here) will be reused.
-          return (
+          return Promise.all([
```

Comment:
Can you talk a bit more about the rationale behind this change? It feels to me that the stats stuff is a little less important than the "basic editing" stuff, so it should be brought in separately in an async call.

But maybe this makes things simpler? Just wondering what you've tried and what the issues were with bringing it in later.

Reply:
So I did this b/c it takes a few minutes for the alert to show up if the state already has an outstanding unresolved answers. In local testing I found it takes 2+ minutes for the graph to "realize" that the stats are actually not-OK.

Doing it immediately here allows the alert to show up as soon as the graph is ready to load. It seems to me this is the most sensical thing to do?

Reply:
Oh, wow -- 2 minutes? Yeah, I agree with you then :)